src/components/form-fields/form-context.tsx

```diff
@@ -0,0 +1,23 @@
+import * as React from 'react';
+import { CatalogItem } from '../../shared/components/catalog/utils/types';
+
+type S = {
+  workspace?: string;
+  application?: string;
+  component?: CatalogItem;
+};
+
+const FormContext = React.createContext<[S, React.Dispatch<React.SetStateAction<S>>]>([
+  {},
+  () => {},
+]);
+
+export const FormContextProvider = ({ children }) => {
+  const [formValues, setFormValues] = React.useState<S>({});
+
+  return (
+    <FormContext.Provider value={[formValues, setFormValues]}>{children}</FormContext.Provider>
```

Comment:
Why do you need a FormContext separately? Doesn't formik provide the same ability to keep form state in context?

Reply:
As discussed on the call, We will use the Formik context and remove FormContext in the follow-up work of journey 1.